### PR TITLE
Closes #1218: Fix turbo mode toggling from Onboarding

### LINKED ISSUE #1218: Turning off Turbo mode in Onboarding doesn't toggle the overlay button
## Steps to reproduce
From fresh install, on the Turbo mode onboarding screen, click 'Turn off'.

### Expected behavior
Turbo mode button in navbar is unselected.

### Actual behavior
Turbo mode button in navbar is still selected until you navigate to a different fragment (ie settings page or Pocket feed) and then re-open the overlay.

### Device information

* Fire TV device: Pendant
* Firefox version: master [`e869530`](https://github.com/mozilla-mobile/firefox-tv/commit/e869530c2be44344627723c83defdf3af395a2be)

### PR DESCRIPTION
**Pull request by psymoon**
### Pull Request checklist

- [x] This PR includes thorough **tests** or an explanation of why it does not
- [x] This PR includes a **CHANGELOG entry** or does not need one

### CODE REVIEW

File: app/src/main/java/org/mozilla/tv/firefox/MainActivity.kt [hunk 1]
```diff
@@ -111,15 +113,26 @@ class MainActivity : LocaleAwareAppCompatActivity(), OnUrlEnteredListener, Media
...
+    // TODO: this is hot fix until NavigationOverlayFragment refactoring is done; we should use LiveData instead
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        if (requestCode == ONBOARDING_REQ_CODE && resultCode == Activity.RESULT_OK) {
+            val fragmentManager = supportFragmentManager
+            val webRenderFragment = fragmentManager.findFragmentByTag(WebRenderFragment.FRAGMENT_TAG) as WebRenderFragment?
+
+            webRenderFragment?.updateOverlayTurboMode()
+
+            if (Settings.getInstance(this@MainActivity).shouldShowPocketOnboarding()) {
```
Comment (liuche) [inline app/src/main/java/org/mozilla/tv/firefox/MainActivity.kt]: This is a workaround, for startActivityForResult being incredibly slow to finish when there is another activity behind. We won't need to worry about this once we can use LiveData for this, so this workaround for now is fine.

**Comment by liuche** [comment]
Oh, didn't see this comment from @dnarcese about TP still being on.

**Comment by psymoon** [comment]
@dnarcese so it looks like Turbo was never set properly from the OverlayActivity (just setting the SharedPreference field, but no on the EngineSession itself)

File: app/src/main/java/org/mozilla/tv/firefox/onboarding/OnboardingActivity.kt [hunk 2]
```diff
@@ -34,10 +34,7 @@ class OnboardingActivity : AppCompatActivity() {
     }
 
     private fun setTurboMode(turboModeEnabled: Boolean) {
-        PreferenceManager.getDefaultSharedPreferences(this)
-                .edit()
-                .putBoolean(Settings.TRACKING_PROTECTION_ENABLED_PREF, turboModeEnabled)
-                .apply()
+        TurboMode.toggle(this, turboModeEnabled)
```
Comment (psymoon) [inline app/src/main/java/org/mozilla/tv/firefox/onboarding/OnboardingActivity.kt]: @dnarcese

Comment (liuche) [reply app/src/main/java/org/mozilla/tv/firefox/onboarding/OnboardingActivity.kt]: Nit: can you change this from `toggle` to `setEnabled`? toggle isn't entirely clear.

**Comment by psymoon** [comment]
@liuche Yes, updating the engine setting is the correct way to apply trackingProtectionPolicy in the app scope. This, however, doesn't update the session directly unless EngineView is re-rendered; therefore, session.enablingTrackingProtection() is needed (maybe @csadilek can clarify further if needed)

**Pull request merged**